### PR TITLE
[NFC] Adding tablegened dialect headers into IMEX_CONFIG_INCLUDE_DIRS

### PR DESCRIPTION
**Pull request by AllanZyne**
Added dialect headers into IMEX_CONFIG_INCLUDE_DIRS, so that other project can build their projects using imex build folder directly, without deploying it (save disk space)

### CODE REVIEW

**Comment by AllanZyne** [comment]
Hi @fschlimb @tkarna, could you help to review this PR? 
Although this PR is not really related to sharpy functionality, but it's really helpful if it can be used it in developping sharpy.

**Comment by fschlimb** [comment]
What does this do exactly? The dir does not exist.

**Comment by fschlimb** [comment]
> ls tools/Imex/include/imex/Dialect/Dist/IR/

The path looks weird and it does not exist in my build:
```
(imex-dev) fschlimb@xxx:~/github/frameworks.ai.mlir.mlir-extensions$ ls build/tools/Imex/include/imex/Dialect/Dist/IR/
ls: cannot access 'build/tools/Imex/include/imex/Dialect/Dist/IR/': No such file or directory
(imex-dev) fschlimb@xxx:~/github/frameworks.ai.mlir.mlir-extensions$ ls build/tools/
CMakeFiles  cmake_install.cmake  imex-cpu-runner  imex-opt  imex-runner
```
But
```
(imex-dev) fschlimb@xxx:~/github/frameworks.ai.mlir.mlir-extensions$ ls build/include/imex/Dialect/Dist/IR/
CMakeFiles  cmake_install.cmake  DistOpsAttrs.cpp.inc  DistOpsAttrs.h.inc  DistOps.cpp.inc  DistOpsDialect.cpp.inc  DistOpsDialect.h.inc  DistOps.h.inc  DistOpsTypes.cpp.inc  DistOpsTypes.h.inc
```

File: cmake/modules/CMakeLists.txt [hunk 1]
```diff
@@ -96,6 +96,7 @@ set(IMEX_CONFIG_INCLUDE_EXPORTS "include(\"\${IMEX_CMAKE_DIR}/IMEXTargets.cmake\
 set(IMEX_CONFIG_INCLUDE_DIRS
     "${IMEX_SOURCE_DIR}/include"
     "${IMEX_BINARY_DIR}/include"
+    "${IMEX_BINARY_DIR}/tools/Imex/include"
```
Comment (silee2) [inline cmake/modules/CMakeLists.txt:99]: This path is only valid when building IMEX as an LLVM external project and setting Imex as the external project name.
I mean, using option 1 in README.md and passing something like
   -DLLVM_EXTERNAL_PROJECTS="Imex" \
   -DLLVM_EXTERNAL_IMEX_SOURCE_DIR=../mlir-extensions

Comment (silee2) [reply cmake/modules/CMakeLists.txt:99]: You would need to change to level CMakeLists.txt
https://github.com/intel/mlir-extensions/blob/main/CMakeLists.txt#L268
To something like.
set(IMEX_GENERATED_HEADER_DIR ${CMAKE_CURRENT_BINARY_DIR})
include_directories(${IMEX_GENERATED_HEADER_DIR}/include)
And then change
https://github.com/intel/mlir-extensions/blob/main/cmake/modules/CMakeLists.txt#L98
to
"${IMEX_GENERATED_HEADER_DIR}/include"

Comment (silee2) [reply cmake/modules/CMakeLists.txt:99]: @AllanZyne Thanks for catching this bug.

**Comment by silee2** [comment]
> What does this do exactly? The dir does not exist.

@fschlimb It is a directory that is created when building IMEX as an LLVM external project. When you build LLVM and IMEX separately build tree structure is different, and 
 that directory is not created.
generated files from LLVM external projects are placed under
`${CMAKE_BINARY_DIR}/tools/<external project name>`
So generated headers for IMEX, if building as an LLVM external project will be located under
`${CMAKE_BINARY_DIR}/tools/Imex/include`
If you are building IMEX separately, generated headers will be in
`${CMAKE_BINARY_DIR}/include`
IMEX_BINARY_DIR == CMAKE_BINARY_DIR in both cases,
The current export logic works correctly for IMEX separate build, but not for IMEX as an LLVM external project. So it is actually a bug.

File: cmake/modules/CMakeLists.txt (hunk 1 above)
Comment (AllanZyne) [reply cmake/modules/CMakeLists.txt]: Thank you for your suggestions, @silee2. Done.

**Review by fschlimb — APPROVED** [review]

**Review by silee2 — APPROVED** [review]

**Pull request merged**